manifests/agent/windows.pp

```diff
@@ -0,0 +1,40 @@
+# Installs the Google Cloud Logging Agent to a Windows machine.
+#
+# For more details please visit:
+#   https://cloud.google.com/logging/docs/agent/installation
+class glogging::agent::windows(
+  $version = 46,
+) {
+
+  $repo = 'https://repo.stackdriver.com/windows'
+  $installer = "StackdriverMonitoring-GCM-${version}.exe"
+  $download_url = "${repo}/${installer}"
+
+  $install_dir = 'C:\Program Files (x86)\Stackdriver\MonitoringAgent'
+  $uninstall_app = "${install_dir}\\uninstall.exe"
+
+  $install_app = "C:\\Windows\\TEMP\\${installer}"
+
+  exec { 'download-stackdriver-agent':
+    command  =>
+      "Invoke-WebRequest -Uri '${download_url}' -OutFile '${install_app}'",
```

Comment:
Might be worth adding proxy support too. https://stackoverflow.com/questions/15646079/invoke-webrequest-getsystemwebproxy

Reply:
PowerShell already supports system proxy settings (I verified on Server 2016 DC -- the one we provide in GCP). Users can override default system proxy using the mechanisms listed in the post you linked to, but it is not necessary (and may conflict with system GPO) if a global proxy is already set.

Given this dilemma I'd say we stay out of the proxy wars and let the user set a proxy if they need one.

Maybe link your post in the docs?

Reply:
Cool, if PS seamlessly picks up system proxy settings, then that sounds perfectly reasonable. I don't know anything about it which prompted me to do a search in case there were additional options that needed to be set at invocation time. I don't know that adding the SO link is necessary since I expect Windows/PS users to know best practices.

Reply:
Agreed.